src/hdskins/java/com/voxelmodpack/hdskins/resources/texture/ThreadDownloadImageEncrypt.java

```diff
@@ -116,36 +112,40 @@ private void loadTexture() {
             case NOPE:
                 LOGGER.debug("Loading http texture from local cache ({})", cacheFile);
                 try {
-                    // e-tag check passed. Load the local file
-                    setLocalCache();
-                    break;
+                    if (setLocalCache()) {// since setLocalCache doesn't throw anything - had to make it return a result
+                        break;
+                    }
                 } catch (IOException e) {
                     // Nope. Local cache is corrupt. Re-download it.
                     // fallthrough to load from network
                     LOGGER.error("Couldn't load skin {}", cacheFile, e);
                 }
-            case OUTDATED:
                 loadTextureFromServer();
         }
     }
 
 
-    private void setLocalCache() throws IOException {
+    private boolean setLocalCache() throws IOException {
         if (Files.isRegularFile(cacheFile)) {
```

Comment:
It looks like this is actually why the cache was never being cleared. The whole thing that set localcache can do nothing and not throw. A redownload would never be triggered in that case.

Reply:
Works perfectly except it's a deprecated class that might not be needed any more...

Killjoy has to confirm whether this works better than whatever he has running at the moment.

Reply:
We're using the vanilla `ThreadDownloadImageData` right now.

Reply:
Yes, but does caching clear correctly now that we're using that? I ask because Poly claims that his fix is "so good we don't even need a manual clear cache button anymore."

Reply:
Speaking of which, I did say I was going to remove tgat, didn't I? :oops:

Reply:
Eh, probably. Honestly it doesn't matter anymore because teh etag is integrated into the file hash now.

Reply:
`Poly claims that his fix is "so good we don't even need a manual clear cache button anymore."`
Not my fix. Your code. The class I gave is a result of Killjoy saying "you can take my re-implementation of vanilla ThreadDownloadImageData, which is ThreadDownloadImageEtag, and re-purpose it for your needs" in DMs.
And all I wanted to ask with this PR (which Solly told me to do instead of just sending him the code) is to look at what I've done. Approve/disapprove/point at obvious mistakes *and help fix them* ~because I'm a noob~/etc.
I just didn't want to create a new file and copy a bunch of code from TDIEtag. In hindsight, I still should've done that, since it looks like TDIEtag is about to be removed entirely.

Reply:
@PolyacovYury Oh, okay. I misunderstood what you were asking then. I thought you'd implemented a change that improved stuff and wanted me to look over it to put into the mod.

Reply:
Here's gotta be a major misunderstanding here (c)

Now that we know what we mean - what will you say?
(Except "close your streams", because a) I heard that, b) I'm not sure what would be the right way to do that).

Reply:
Also - this convo can be moved to my fork's repo (https://github.com/PolyacovYury/MineLittlePony/pull/1).